game/magic/game/game.go

```diff
@@ -596,6 +598,75 @@ func (game *Game) FindValidCityLocationOnContinent(x int, y int) (int, int) {
     return 0, 0
 }
 
+func (game *Game) SuggestCityName(race data.Race) (string) {
+    allCities := game.AllCities()
+    fallback := fmt.Sprintf("City %d", len(allCities)+1)
+
+    predefinedNames, err := citylib.ReadCityNames(game.Cache)
+    if err != nil {
+        log.Printf("Unable to read predefined city names: %v", err)
+        return fallback
+    }
+
+    if len(predefinedNames) % 14 != 0 {
+        log.Printf("Predefined city names not dividable by number of races")
+        return fallback
+    }
+
+    raceIndex := 0
+    switch race {
+        case data.RaceBarbarian: raceIndex = 0
+        case data.RaceBeastmen: raceIndex = 1
+        case data.RaceDarkElf: raceIndex = 2
+        case data.RaceDraconian: raceIndex = 3
+        case data.RaceDwarf: raceIndex = 4
+        case data.RaceGnoll: raceIndex = 5
+        case data.RaceHalfling: raceIndex = 6
+        case data.RaceHighElf: raceIndex = 7
+        case data.RaceHighMen: raceIndex = 8
+        case data.RaceKlackon: raceIndex = 9
+        case data.RaceLizard: raceIndex = 10
+        case data.RaceNomad: raceIndex = 11
+        case data.RaceOrc: raceIndex = 12
+        case data.RaceTroll: raceIndex = 13
+    }
+
+    var existingNames []string
+    for _, city := range allCities {
+        existingNames = append(existingNames, city.Name)
+    }
+
+    // try to find a unused city name
+    entriesPerRace := len(predefinedNames) / 14
+    for _, i := range rand.Perm(entriesPerRace) {
+        name := predefinedNames[entriesPerRace * raceIndex + i]
+        if !slices.Contains(existingNames, name) {
+            return name
+        }
+    }
+    
+    // try to find a used city name without any number
+    expression := regexp.MustCompile(`(.*) (\d+)$`)
+    var matches []string
+    for _, i := range rand.Perm(entriesPerRace) {
+        name := predefinedNames[entriesPerRace * raceIndex + i]
```

Comment:
I think theres a bug here in that the regexp shouldn't match the predefined name, since the predefined name will always just be `somecity` and not `somecity 2`. The regexp should match over all the existingNames instead. I'll just implement a simple loop instead.